examples/rgw_admin_curl.sh

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env bash
+#
+# Copyright (C) 2022 Red Hat <contact@redhat.com>
+#
+# Author: Prashant D <pdhange@redhat.com>
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Library Public License as published by
+# the Free Software Foundation; either version 2, or (at your option)
+# any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Library Public License for more details.
+#
+
+show_help()
+{
+    echo "Usage: `basename $0` -a <access-key> -s <secret-key>" \
+                 "-e <rgw-endpoint> -r <http-request>" \
+                 "-p <admin-resource> -q \"<http-query-string>\""
+    echo "   -a       Access key of rgw user"
+    echo "   -s       Secret key of rgw user"
+    echo "   -e       RGW endpoint in <ipaddr:port> format"
+    echo "   -r       HTTP request type GET/PUT/DELETE"
+    echo "   -p       RGW admin resource e.g user, bucket etc"
+    echo "   -q       HTTP query string"
+    echo "   e.g"
+    echo "      # `basename $0` -a ABCD1234EFGH5678IJ90" \
+                "-s klmnopqrstuvwxyz12345ABCD987654321efghij" \
+                "-e 10.0.0.1:8080 -r GET -p user -q \"uid=admin\""
+    exit 1
+}
+
+access_key=""
+secret_key=""
+rgw_endpoint=""
+http_request=""
+admin_resource=""
+http_query=""
+
+while getopts "a:s:e:r:p:q:" opt; do
+    case "$opt" in
+    a)
+        access_key=${OPTARG}
+        ;;
+    s)  secret_key=${OPTARG}
+        ;;
+    e)  rgw_endpoint=${OPTARG}
+        ;;
+    r)  http_request=${OPTARG}
+        ;;
+    p)  admin_resource=${OPTARG}
+        ;;
+    q)  http_query=${OPTARG}
+        ;;
+    *)
+        show_help
+        exit 1
+        ;;
+    esac
+done
+shift $((OPTIND-1))
+
+if [ -z "${access_key}" ] || [ -z "${secret_key}" ] || \
+   [ -z "${rgw_endpoint}" ] || [ -z "${http_request}" ] || \
+   [ -z "${admin_resource}" ] || [ -z "${http_query}" ]; then
+    show_help
+else
+    echo "Processed args : access_key=${access_key}, secret_key=${secret_key}, " \
+         "rgw_endpoint=${rgw_endpoint}, http_request=${http_request}, " \
+         "admin_resource=${admin_resource}, http_query=${http_query}"
+fi
+
+resource="/admin/${admin_resource}"
+contentType="application/x-compressed-tar"
+dateTime=`date -R -u`
+
+headerToSign="${http_request}
+
+${contentType}
+${dateTime}
+${resource}"
+
+signature=`echo -en "$headerToSign" | \
+           openssl sha1 -hmac ${secret_key} -binary | base64`
+
+echo "Sending ${http_request} request to ${rgw_endpoint} :" 
+curl -X ${http_request} -H "Content-Type: ${contentType}" -H "Date: ${dateTime}" \
+     -H "Authorization: AWS ${access_key}:${signature}" -H "Host: ${rgw_endpoint}" \
+     "http://${rgw_endpoint}${resource}?${http_query}" 2> /dev/null|jq "."
```

Comment:
i feel like this script would be more flexible if it didn't pipe to `jq .`. users may not have jq installed, or they may want to do their own filtering on top

to support that, we'd want to remove the `echo`s too. what do you think?

Reply:
Yes. Makes sense. Let me remove jq part from default behavior and add option to use jq if specified.  Removing `echo` s too.